### PR TITLE
Updates Test Suite To Use New GA API Plus Adds Missing Functionality To Our Internal Utility Clients

### PR DESCRIPTION
**Pull request by safwank97**
### Which issue this PR addresses:

Fixes https://issues.redhat.com/browse/ARO-4618

### What this PR does / why we need it:

This PR updates our e2e test suite to use the new GA API version `20231122`, instead of the `20230701preview` version of the API. Plus it also focuses to update our internal, manually-written wrappers around Azure SDK clients and adds back the missing `UpdateAndWait()` functionality to `20231122` and `20240812preview`. This will help us to create clusters with this new GA API version. 

### Test plan for issue:

N/A.

### Is there any documentation that needs to be updated for this PR?

None that I'm aware of.

### CODE REVIEW

File: pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go [hunk 1]
```diff
@@ -12,6 +12,7 @@ import (
 // OpenShiftClustersClientAddons contains addons for OpenShiftClustersClient
 type OpenShiftClustersClientAddons interface {
 	CreateOrUpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230904.OpenShiftCluster) error
+	UpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230904.OpenShiftClusterUpdate) error
```
Comment (tony-schndr) [inline pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go:15]: We don't need to add this, these clients are only used internally, nothing would call this.

File: pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go [hunk 2]
```diff
@@ -26,6 +27,15 @@ func (c *openShiftClustersClient) CreateOrUpdateAndWait(ctx context.Context, res
 	return future.WaitForCompletionRef(ctx, c.Client)
 }
 
+func (c *openShiftClustersClient) UpdateAndWait(ctx context.Context, resourceGroupName string, resourceName string, parameters mgmtredhatopenshift20230904.OpenShiftClusterUpdate) error {
```
Comment (tony-schndr) [inline pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go:30]: same as above

**Review by tony-schndr — COMMENTED** [review]
This isn't really a HOTFIX right?  This is just the internal client we *can* use in `util/`, `e2e/`, etc...  I'd also say we don't need to add this to any other client that doesn't need it.
I would also say we could save some time and effort here and just include the necessary client addons in the PR when cutting over e2e to use the new API.

File: pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go (hunk 1 above)
Comment (safwank97) [reply pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go:15]: Yes that makes sense I'll remove them from `20230904` but will keep on other two where it is needed `20231122` and `20240812preview`.

Comment (safwank97) [reply pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go]: Fixed U+1F44D

File: pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go (hunk 2 above)
Comment (safwank97) [reply pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go]: Fixed

File: pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go (hunk 1 above)
Comment (safwank97) [reply pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go]: Plus the question with this pr or to include it with e2e test suite work, I have another branch for my e2e test suite work which I could bump it off after this PR later with the help of rebase. If it's fine with everyone.

Comment (safwank97) [reply pkg/util/azureclient/mgmt/redhatopenshift/2023-09-04/redhatopenshift/openshiftclusters_addons.go]: No worries made this whole pr focusing towards e2e test suite work plus including the missing functionality to our internal utility clients. Basically all in one pr. Therefore closing this conversation with resolved mark.

**Review by tsatam — COMMENTED** [review]
Changes LGTM. I'd like to see a passing E2E run before merge, though.

**Review by tony-schndr — APPROVED** [review]
Nice work Mohammed!

**Comment by safwank97** [comment]
> Nice work Mohammed!

Thank you :)

**Review by cadenmarchese — APPROVED** [review]
lgtm!

**Review by shubhadapaithankar — APPROVED** [review]
LGTM

**Pull request merged**